src/Interpreter.cpp

```diff
@@ -107,6 +115,70 @@ RamDomain Interpreter::evalVal(const RamValue& value, const InterpreterContext&
             }
         }
 
+        RamDomain visitUserDefinedOperator(const RamUserDefinedOperator& op) override {
+            // get name and type
+            const std::string& name = op.getName();
+            const std::string& type = op.getType();
+
+            // load DLL (if not done yet)
+            void* handle = interpreter.loadDLL();
+            void (*fn)() = (void (*)())dlsym(handle, name.c_str());
+            if (fn == nullptr) {
+                std::cerr << "Cannot find user-defined operator " << name << " in " << SOUFFLE_DLL
+                          << std::endl;
+                exit(1);
+            }
+
+            // prepare dynamic call environment
+            size_t n = op.getArgNum();
```

Comment:
What is 'n', 'cif', 'rc'? These names are far from clear

Reply:
This makes only sense in the context of the ffi library. See standard example there.